File: Introduction/7__array_n_elements.scala

// https://www.hackerrank.com/challenges/fp-array-of-n-elements

// def f(num:Int) : List[Int] = {
// 	for (x <- (0 until num)) yield x
// }

List.fill(num)(num)
List.range(1, num)

// def f(num:Int) : List[Int] = {
//     num match {
//         case 0 => Nil
//         default => num :: f(num-1)
//     }
// }

// def f(num:Int) : List[Int] = if(num==0) Nil else num::f(num-1)

File: Introduction/7__array_n_elements.hs

-- https://www.hackerrank.com/challenges/fp-array-of-n-elements

fn n    | n == 0 = []
        | otherwise = fn (n-1) ++ [(n-1)]
        -- | otherwise = (n-1):  fn (n-1)

fn n = [h | h <- [1..n]]

fn n = [1..n]